libraries/render-utils/src/Model.cpp

```diff
@@ -958,10 +958,10 @@ void Model::setCauterized(bool cauterized, const render::ScenePointer& scene) {
     }
 }
 
-void Model::setPrimitiveMode(PrimitiveMode primitiveMode) {
+void Model::setPrimitiveMode(PrimitiveMode primitiveMode, const render::ScenePointer& scene) {
     if (_primitiveMode != primitiveMode) {
         _primitiveMode = primitiveMode;
-        setRenderItemsNeedUpdate();
+        updateRenderItemsKey(scene);
```

Comment:
```suggestion
        updateRenderItemsKey(nullptr);
```
And remove the second parameter from `setPrimitiveMode()`?